linker/Mono.Linker/Driver.cs

```diff
@@ -294,6 +296,7 @@ static void Usage (string msg)
 			Console.WriteLine ("                 default is all");
 			Console.WriteLine ("   -x          Link from an XML descriptor");
 			Console.WriteLine ("   -a          Link from a list of assemblies");
+			Console.WriteLine ("   -r          Link from a list of assemblies using public roots");
```

Comment:
Sure, but this can break with -a foo.exe (where just Main is used as an entry point) as well if you remove fields from an argument to Marshal.SizeOf. And, of course reflection can also make field removal breaking. -r is just another option that can be supplemented with extra roots in xml files or extra plugin analyses.